server/src/main/java/org/elasticsearch/search/aggregations/metrics/cardinality/CardinalityAggregator.java

```diff
@@ -181,6 +200,48 @@ public void close() {
         }
     }
 
+    // FIXME: This "RollupCollector" was written as a hack to prove that deserializing the HLLs might work
+    private static class RollupCollector extends Collector {
+
+        private final HyperLogLogPlusPlus counts;
+        private final SortedBinaryDocValues rollups;
+
+        RollupCollector(HyperLogLogPlusPlus counts, SortedBinaryDocValues rollups) {
+            this.counts = counts;
+            this.rollups = rollups;
+        }
+
+        @Override
+        public void collect(int doc, long bucketOrd) throws IOException {
+            // Each binary blob is in the SortedBinaryDocValues object, so we just advance along deserialize, and merge.
+            if (rollups.advanceExact(doc)) {
+                BytesRef bytes = rollups.nextValue();
+                byte[] hllBytes = bytes.bytes;
+                assert hllBytes.length > 0 : "Decoded HLL had no bytes";
+                ByteArrayInputStream bais = new ByteArrayInputStream(hllBytes);
+                InputStreamStreamInput issi = new InputStreamStreamInput(bais);
+                HyperLogLogPlusPlus rollup = HyperLogLogPlusPlus.readFrom(issi, BigArrays.NON_RECYCLING_INSTANCE);
+                long cardCounts1 = counts.cardinality(0);
+                long cardRollups1 = rollup.cardinality(0);
+                counts.merge(0, rollup, 0);
+                long cardCounts2 = counts.cardinality(0);
+                long cardRollups2 = rollup.cardinality(0);
+                long maxBucket = counts.maxBucket();
```

Comment:
The above `long` values are just for debugging and will come out once we think on this section a bit more.